src/firewall.c

```diff
@@ -89,29 +89,40 @@ unsigned int FW_MARK_MASK;             /**< @brief Iptables mask: bitwise or of
 char *
 arp_get(const char req_ip[])
 {
-	FILE *proc;
-	char ip[INET6_ADDRSTRLEN];
-	char mac[18];
+	int type = 0;
+	int flags = 0;
+	char ip[16] = {0};
+	char mac[18] = {0};
+	char mask[16] = {0};
+	char line[128] = {0};
+	char dev[128] = {0};
 	char *reply = NULL;
+	FILE *proc_arp = NULL;
 
-	if (!(proc = fopen("/proc/net/arp", "r"))) {
+	if (!(proc_arp = fopen("/proc/net/arp", "r"))) {
 		return NULL;
 	}
 
 	/* Skip first line */
-	while (!feof(proc) && fgetc(proc) != '\n');
+	fgets(line, sizeof(line), proc_arp);
 
 	/* Find ip, copy mac in reply */
-	reply = NULL;
-	while (!feof(proc) && (fscanf(proc, " %15[0-9.] %*s %*s %17[A-Fa-f0-9:] %*s %*s", ip, mac) == 2)) {
-		if (strcmp(ip, req_ip) == 0) {
+	while (fgets(line, sizeof(line), proc_arp))
+	{
+		if (sscanf(line, "%s 0x%x 0x%x %s %s %s\n",
```

Comment:
/proc/net/arp is in read only flash and is just a place holder. Arp is a function of dnsmasq in OpenWrt and the arp command will return the arp table that is held in ram by dnsmasq but this is the same data (for lan connected devices) as stored in dhcp.leases.
I did a quick test here, on Chaos Calmer, with a device with a fixed ip address and get the same syslog error as you do, yet the ip/mac is listed by the arp command. Nodogsplash does not issue a token. I have always considered this as a correct function, without looking into it, as access to the Internet through a Hotspot controller using a static ip address is not desirable. Are you using a static IP?